### PR TITLE
Differentiate between empty or no fingerprint

### PR DESCRIPTION
**Pull request by koplas**
This pull request introduces an API change, which allows for a better diagnosis of fingerprint errors.

### CODE REVIEW

File: cmd/csaf_checker/processor.go [hunk 1]
```diff
@@ -1518,8 +1518,13 @@ func (p *processor) checkPGPKeys(_ string) error {
...
 		}
 
+		if key.Fingerprint == "" {
+			p.badPGPs.warn("No fingerprint for public OpenPGP key found.")
+			continue
+		}
+
 		if !strings.EqualFold(ckey.GetFingerprint(), string(key.Fingerprint)) {
-			p.badPGPs.error("Fingerprint of public OpenPGP key %s does not match remotely loaded.", u)
+			p.badPGPs.error("Given Fingerprint ('%q') of public OpenPGP key %q does not match remotely loaded ('%q').", string(key.Fingerprint), u, ckey.GetFingerprint())
```
Comment (s-l-teichmann) [inline cmd/csaf_checker/processor.go:1527]: `s/'%q'/%q/g`

**Review by s-l-teichmann — CHANGES_REQUESTED** [review]

**Comment by tschmidtb51** [comment]
Is the case "no fingerprint is given for this key" differentiated from "an empty fingerprint is given for this key"?

**Comment by koplas** [comment]
> Is the case "no fingerprint is given for this key" differentiated from "an empty fingerprint is given for this key"?

No, in this implementation, both cases are handled the same. To separate both cases, `csaf.PGPKey` needs to be changed, which affects the public API.

**Comment by tschmidtb51** [comment]
> > Is the case "no fingerprint is given for this key" differentiated from "an empty fingerprint is given for this key"?
> 
> No, in this implementation, both cases are handled the same. To separate both cases, `csaf.PGPKey` needs to be changed, which affects the public API.

In future versions, we should differentiate that. 
@bernhardreiter / @s-l-teichmann: Please decide when it makes most sense to implemented this.

**Comment by bernhardreiter** [comment]
> > Is the case "no fingerprint is given for this key" differentiated from "an empty fingerprint is given for this key"?

> To separate both cases, `csaf.PGPKey` needs to be changed, which affects the public API.

Then we should not implement it for 3.x as we would need to go to version 4 if we change the public API. @koplas can you implement a version that just gives more details for 3.x? (Please do not put in unrelated whitespace changes.)

@tschmidtb51 wrote:
> In future versions, we should differentiate that.

If I understand the schema correctly there is a ` "minLength": 40` in place. So I'd expect an empty fingerprint value would violate the schema. So I expect both cases to result into to different messages. @koplas can you test what happens with an empty fingerprint value in the current code?

**Comment by bernhardreiter** [comment]
(@s-l-teichmann sorry for the back and forth of your review. It has been addressed and I am trying to remove you as a requested reviewer, which somehow does not work.)

**Comment by koplas** [comment]
> If I understand the schema correctly there is a ` "minLength": 40` in place. So I'd expect an empty fingerprint value would violate the schema. So I expect both cases to result into to different messages. @koplas can you test what happens with an empty fingerprint value in the current code?

If the fingerprint is empty, the `csaf_checker` prints the following error message: "Could not parse the Provider-Metadata.json".
Using the `csaf_downloader` will result in "no valid provider-metadata.json found".
No additional error messages are printed with this branch and the schema validation prevents the check for empty fingerprints.

**Comment by bernhardreiter** [comment]
As discussed in #555 we won't be using this approach.

**Pull request closed without merging**